src/runtime/objmodel.cpp

```diff
@@ -677,6 +678,11 @@ void Box::setattr(const std::string& attr, Box* val, SetattrRewriteArgs* rewrite
         return;
     }
 
+    if (cls->tp_setattro) {
+        cls->tp_setattro(this, boxString(attr), val);
+        return;
+    }
+
```

Comment:
quite right.  Didn't I do this exact same thing with the `__getattr__` changes?  removed.